### PR TITLE
Maze generation

### PR DESCRIPTION
**Pull request by SineNomine2001**

### CODE REVIEW

File: Project48/Assets/Scripts/MapGeneration/MazeRenderer.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+
+public class MazeRenderer : MonoBehaviour
```
Comment (cuoshen) [inline Project48/Assets/Scripts/MapGeneration/MazeRenderer.cs:6]: If you want to communicate with existing code I suggest you add namespace Jail, or remove all namespace from existing code.

**Review by cuoshen — COMMENTED** [review]
Overall LGTM, still need to fix a few things.

File: Project48/Assets/Scripts/MapGeneration/MazeRenderer.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            Vector2Int startPos = new Vector2Int(rng.Next(width), 0);
+            Vector2Int endPos = new Vector2Int(rng.Next(width), height - 1);
+
+            DrawRBMaze(lastEndPos, startPos, endPos, maze);
+            lastEndPos += new Vector3(endPos.x - startPos.x, 0, endPos.y - startPos.y + 1);
+            yield return null; // Physics.CheckSphere cannot detect colliders generated in the same tick
+        }
+    }
+
+    void DrawRBMaze(Vector3 lastEndPos, Vector2Int startPos, Vector2Int endPos, WallState[,] maze)
```
[the remark this answers is not in the document]
Comment (SineNomine2001) [reply Project48/Assets/Scripts/MapGeneration/MazeRenderer.cs:41]: Duh

File: Project48/Assets/Scripts/MapGeneration/MazeGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            GenerateRBMaze(lastEndPos, startPos, endPos, maze);
+            lastEndPos += new Vector3(endPos.x - startPos.x, 0, endPos.y - startPos.y + 1);
+            yield return null; // Physics.CheckSphere cannot detect colliders generated in the same tick
+        }
+    }
+
+    /// <summary>
+    /// Generate an individual rectangular RB maze, its entrance connected to the exit of the previous maze
+    /// </summary>
+    /// <param name="lastEndPos"></param> Global position of the exit of previous maze
```
Comment (cuoshen) [inline Project48/Assets/Scripts/MapGeneration/MazeGenerator.cs:44]: Misplaced comments

**Review by cuoshen — COMMENTED** [review]
Merge after fixing comment

File: Project48/Assets/Scripts/MapGeneration/MazeGenerator.cs (hunk 3 above)
Comment (SineNomine2001) [reply Project48/Assets/Scripts/MapGeneration/MazeGenerator.cs:44]: Duh Duh

**Pull request merged**